introduce type=measurement and migrate 48 fields to use it (#1988)
openstreetmap/id-tagging-schema: SCHEMA.md

An optional property to reference to the icons of another field, indicated  by using that field's name contained in brackets, like `{field}`. This is for example useful when there are multiple variants of fields for the same tag, which should all use the same icons.

##### `measurement`

> Used when `type = measurement`. Defines the unit of measurements that are supported by this field. For example:

Reviewer: ```suggestion
Used when `type = measurement` defines the unit of measurements that are supported by this field. For example:
```

seems to flow better for me

disclaimer: not a native speaker

```json
{
  "key": "diameter",
  "type": "measurement",
  "measurement": {
    // The dimension being measured. This constrains the permitted units.

>     // The ID id defined by CLDR.

Reviewer: maybe link CLDR in description above?

at least I have not really understood what CLDR even is before

or is it case "obviously google for it if unclear"?

>     // The ID id defined by CLDR.

Reviewer: "The ID id"?

    "dimension": "length",

    // The corresponding 'usage' from CLDR.
    "usage": "default",

>     // If the field only allows some units, you can list them here

Reviewer: same as earlier in .json

Author: corresponds to https://github.com/openstreetmap/id-tagging-schema/pull/1988#discussion_r3638968548

    // using CLDR's unit names. If not specified, then, all units from
    // this dimension are allowed.
    "units": ["meter", "centimeter", "foot-and-inch"],

>     // Some OSM tags have a default unit, which does not need to be explicitly included in the tag value.

Reviewer: same as earlier in .json

Author: sorry, which earlier comment? there are 59 comments now 😆

Reviewer: https://github.com/openstreetmap/id-tagging-schema/pull/1988#pullrequestreview-4765187268

>     // This field defines how to interpret a unit-less value:

Reviewer: ```suggestion
    // This field defines how to interpret an implicit unit:
```

```suggestion
    // This field defines how to interpret case without explicit unit:
```

```suggestion
    // This field defines how to interpret case where unit is not specified:
```

whole fun is that it is not unitless.

    "impliedUnit": "meter"
  }
}
```

To convert the unit IDs into the values used by OSM, see [§Units](#Units).

Translations for the [`narrow` and `long` form](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat) of each unit are bundled into iD-tagging-schema's locale files.

### Deprecations

> Use `deprecated.json` ([Example](https://github.com/openstreetmap/id-tagging-schema/blob/main/data/deprecated.json), [Schema](https://github.com/ideditor/schema-builder/blob/main/schemas/deprecated.json)) to specify tag deprecations.

Reviewer: migration guide file also should get an entry, right?

Reviewer: maybe include there (or in docs) some high-level overview how that thing works? Sadly, I am still not entirely sure and mostly scared by this
